From fa8be7be9457028a9ac5d832d1710263594dbc21 Mon Sep 17 00:00:00 2001
From: Robert Marko <robert.marko@sartura.hr>
Date: Wed, 4 Nov 2020 20:36:53 +0100
Subject: [PATCH] arm64: dts: ipq8074: add SPMI regulators

IPQ8074 has PMD9655 regulators connected via SPMI.
These are supported by SPMI regulator driver and needed for
CPU, UBI scaling as well as SD/eMMC.

Signed-off-by: Robert Marko <robert.marko@sartura.hr>
---
 arch/arm64/boot/dts/qcom/ipq8074.dtsi | 46 +++++++++++++++++++++++++++
 1 file changed, 46 insertions(+)

--- a/arch/arm64/boot/dts/qcom/ipq8074.dtsi
+++ b/arch/arm64/boot/dts/qcom/ipq8074.dtsi
@@ -5,6 +5,7 @@
 
 #include <dt-bindings/interrupt-controller/arm-gic.h>
 #include <dt-bindings/clock/qcom,gcc-ipq8074.h>
+#include <dt-bindings/spmi/spmi.h>
 
 / {
 	model = "Qualcomm Technologies, Inc. IPQ8074";
@@ -136,6 +137,15 @@
 		};
 	};
 
+	e_smps1_reg: fixed-regulator@0 {
+		compatible = "regulator-fixed";
+		regulator-name = "e-smps1-reg";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		regulator-always-on;
+		regulator-boot-on;
+	};
+
 	soc: soc {
 		#address-cells = <0x1>;
 		#size-cells = <0x1>;
@@ -807,6 +817,42 @@
 			#size-cells = <0>;
 			interrupt-controller;
 			#interrupt-cells = <4>;
+
+			pmic@1 {
+				compatible ="qcom,spmi-pmic";
+				reg = <0x1 SPMI_USID>;
+				#address-cells = <1>;
+				#size-cells = <0>;
+
+				regulators {
+					compatible = "qcom,pmd9655-regulators";
+					vdd_s3-supply = <&e_smps1_reg>;
+					vdd_s4-supply = <&e_smps1_reg>;
+					vdd_ldo11-supply = <&e_smps1_reg>;
+
+					s3: s3 {
+						regulator-name = "pmd9655_s3";
+						regulator-min-microvolt = <592000>;
+						regulator-max-microvolt = <1064000>;
+						regulator-always-on;
+						regulator-boot-on;
+					};
+
+					s4: s4 {
+						regulator-name = "pmd9655_s4";
+						regulator-min-microvolt = <712000>;
+						regulator-max-microvolt = <992000>;
+						regulator-always-on;
+						regulator-boot-on;
+					};
+
+					ldo11: ldo11 {
+						regulator-name = "pmd9655_ldo11";
+						regulator-min-microvolt = <1104000>;
+						regulator-max-microvolt = <3300000>;
+					};
+				};
+			};
 		};
 
 		apcs_glb: mailbox@b111000 {
